stdlib/format.ml

```diff
@@ -314,23 +272,22 @@ let break_same_line state width =
    beyond pp_max_indent, then the box is rejected on the left
    by simulating a break. *)
 let pp_force_break_line state =
-  match state.pp_format_stack with
-  | Format_elem (bl_ty, width) :: _ ->
+  match Stack.top_opt state.pp_format_stack with
+  | None -> pp_output_newline state
+  | Some { box_type; width } ->
     if width > state.pp_space_left then
-      (match bl_ty with
-       | Pp_fits -> () | Pp_hbox -> ()
-       | Pp_vbox | Pp_hvbox | Pp_hovbox | Pp_box ->
-         break_line state width)
-  | [] -> pp_output_newline state
+      match box_type with
+      | Pp_fits | Pp_hbox -> ()
+      | Pp_vbox | Pp_hvbox | Pp_hovbox | Pp_box -> break_line state width
 
 
 (* To skip a token, if the previous line has been broken. *)
 let pp_skip_token state =
-  (* When calling pp_skip_token the queue cannot be empty. *)
-  match take_queue state.pp_queue with
-  | { elem_size = size; length = len; token = _; } ->
-    state.pp_left_total <- state.pp_left_total - len;
-    state.pp_space_left <- state.pp_space_left + int_of_size size
+  match Queue.take_opt state.pp_queue with
+  | None -> () (* print_if_newline must have been the last printing command *)
```

Comment:
Previous comment indicates that this case cannot happen; new comment indicates that it can. Could you say a few words to understand the discrepancy ?

Reply:
I had some explanation in the [commit message](https://github.com/ocaml/ocaml/pull/1948/commits/a8b43c7ea39b491cab3ff3c561bc337d14c74a49):

> Function `pp_skip_token` had a comment saying that the queue is never empty when it is called, and it was tempting to rely on that. However, I came up with a unit test that falsifies that invariant, see `pp_skip_token.ml`. This prompted to change `pp_skip_token` to gracefully handle the case when the queue is empty. Before, the invariant was wrong, but the code still worked correctly because the exception would have been caught in `advance_left`.

Reply:
Thanks, I had missed that!